apps/desktop-wallet/src/api/apiDataHooks/wallet/useFetchWalletWorth.ts

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import useFetchListedFtsWorth from '@/api/apiDataHooks/utils/useFetchListedFtsWorth'
+import useFetchSeparatedTokensByListing from '@/api/apiDataHooks/utils/useFetchSeparatedTokensByListing'
+import useMergeAllTokensBalances from '@/api/apiDataHooks/utils/useMergeAllTokensBalances'
+import useFetchWalletBalancesAlph from '@/api/apiDataHooks/wallet/useFetchWalletBalancesAlph'
+import useFetchWalletBalancesTokens from '@/api/apiDataHooks/wallet/useFetchWalletBalancesTokens'
+
+const useFetchWalletWorth = () => {
+  const { data: alphBalances, isLoading: isLoadingAlphBalances } = useFetchWalletBalancesAlph()
+  const { data: tokensBalances, isLoading: isLoadingTokensBalances } = useFetchWalletBalancesTokens()
+  const allTokensBalances = useMergeAllTokensBalances({
+    includeAlph: true,
+    alphBalances,
+    tokensBalances
+  })
+  const {
+    data: { listedFts },
+    isLoading: isLoadingTokensByListing
+  } = useFetchSeparatedTokensByListing(allTokensBalances)
+  const { data: worth, isLoading: isLoadingWorth } = useFetchListedFtsWorth({ listedFts })
```

Comment:
If `allTokensBalances` isn't available, it will be `[]`, meaning `listedFts` will be `[]`, meaning that `useFetchListedFtsWorth` will be disabled as an effect.